spec/models/report_spec.rb

```diff
@@ -0,0 +1,82 @@
+require "rails_helper"
+
+RSpec.describe Report, type: :model do
+  describe "Enums" do
+    it { is_expected.to define_enum_for(:status).with_values([:waiting, :checked, :rejected]) }
+  end
+
+  describe "Associations" do
+    it { is_expected.to have_many(:comments).dependent(:destroy) }
+    it { is_expected.to belong_to(:user) }
+  end
+
+  describe "Validations" do
+    context "when all fields given" do
+      let(:report) {FactoryBot.create :report}
```

Comment:
let khác gì let! e ?

Reply:
khi khai báo let thì nó chỉ có giá trị trong mỗi block đó.
Khi khai báo let! thì nó có giá trị trong mọi block it của nó. Bản chất của nó là một khối before ngầm nên nó sẽ được tính toán lại trước mỗi block it.